addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -170,7 +170,7 @@ function ThreadFactory({ Entity }) {
          */
         static async joinChannel(channelId, { autoselect = false } = {}) {
             const channel = this.channelFromId(channelId);
-            if (channel) {
+            if (channel && channel.isPinned) {
```

Comment:
Isn't that a programming error to reach here if isPinned?

Reply:
Old code made assumption that all channels in mail manager have been joined by current user.
It also forces all channels being pinned on UI.

This worked because it only added channels in mail manager when actually joined.
Now with current store, a channel may already exist in store even though current user hasn't joined it yet. For instance, if a message makes reference to channel with ID 5 that we are not yet aware of, it just creates an entity for this channel.

This is still not perfect, because this code assumes "pinned = joined", but that's still much better than before.